src/Microsoft.ML.Recommender/MatrixFactorizationTrainer.cs

```diff
@@ -26,11 +26,33 @@ namespace Microsoft.ML.Trainers
     /// Train a matrix factorization model. It factorizes the training matrix into the product of two low-rank matrices.
     /// </summary>
     /// <remarks>
-    /// <para>The basic idea of matrix factorization is finding two low-rank factor marcies to apporimate the training matrix.
-    /// In this module, the expected training data is a list of tuples. Every tuple consists of a column index, a row index,
+    /// <format type="text/markdown"><![CDATA[
+    /// To create this trainer, use [MatrixFactorization](xref:Microsoft.ML.RecommendationCatalog.RecommendationTrainers.MatrixFactorization(System.String,System.String,System.String,System.Int32,System.Double,System.Int32))
+    /// or [MatrixFactorization(Options)](xref:Microsoft.ML.RecommendationCatalog.RecommendationTrainers.MatrixFactorization(Microsoft.ML.Trainers.MatrixFactorizationTrainer.Options)).
+    ///
+    /// ### Input and Output Columns
+    /// There are three input columns required, one for matrix row indexes, one for matrix column indexes, and one for values (i.e., labels) in matrix .
+    /// They together defines a matrix in [COO](https://en.wikipedia.org/wiki/Sparse_matrix#Coordinate_list_(COO)) format.
+    /// The type of label column is vector of <xref:System.Single> while the other two columns are key-typed scalar.
```

Comment:
>key-typed scalar [](start = 99, length = 16)

Key-typed scalar? What does that mean? Are you talking about KeyType? Should this have an xref?

Reply:
Will do `[key-typed](<xref:Microsoft.ML.Data.KeyDataViewType>) scalar`

---
In reply to: [276858813](https://github.com/dotnet/machinelearning/pull/3409#discussion_r276858813) [](ancestors = 276858813)